internal/pkg/agent/application/gateway/fleet/fleet_gateway.go

```diff
@@ -124,7 +124,7 @@ func newFleetGatewayWithScheduler(
 		acker:        acker,
 		stateFetcher: stateFetcher,
 		stateStore:   stateStore,
-		errCh:        make(chan error),
+		errCh:        make(chan error, 1),
```

Comment:
Same here? The `Errors()` should be watched by the coordinator, so why does this need to be changed to buffered. What is causing this to be blocking unless you make it buffered.

Reply:
Fixed managed_mode coordination with fleet gateway. Now the gateway errors reading loop waits until gateway exits. Otherwise if the gateway shuts down out of sequence blocks it can block on errCh.